### ISSUE TITLE
Export and Import trained models

### ISSUE THREAD

**Issue by kokhazade** [oneapi-src/oneDNN#1136]
Hi,
I want to train a model in a program and use it in another (the first program is for training and second one is for inference). How should I export or save the trained model with its parameters in the first program? How should I import the mentioned saved model to the the second program for inference?
I want to use a same model in train and inference phases but I want to have train and inference in distinct programs.

**Comment by igorsafo** [comment]
Hi @kokhazade ,
oneDNN is a library that provides optimized building blocks for DNN applications and it doesn't contain functionality to load/store model parameters. Usually such functionality is implemented in frameworks as part of their core functionality. Each framework has its own format for parameters. For example:

- PyTorch: https://pytorch.org/tutorials/beginner/basics/saveloadrun_tutorial.html
- TensorFlow: https://www.tensorflow.org/tutorials/keras/save_and_load
- OpenVino: https://docs.openvinotoolkit.org/latest/workbench_docs_Workbench_DG_Select_Models.html

**Comment by kokhazade** [comment]
@igorsafo Thanks for your response.
So, how can I use oneDNN library in my application?
For example I want to train an Alexnet and use the results in inference mode.
I want to train model and then feed an image and see the result of inference.
How can I do it?

**Comment by igorsafo** [comment]
@kokhazade oneDNN is used to replace framework implementations of DNN operations. For example, in case of Alexnet, Convolution, Relu, Pooling, Inner Product (Fully-connected) and LRN operations can be replaced with oneDNN primitives on application side. Here is an example of part of Alexnet implemented using oneDNN primitives: https://github.com/oneapi-src/oneDNN/blob/master/examples/cnn_training_f32.cpp

Again, oneDNN contains only part of all operations required for Inference/Training. It doesn't contain utility functions like Dropout, Loss, SGD solver, exporting/saving model, etc. If you plan to train many models and want to experiment with them then the best way would be to work with popular frameworks (TensorFlow, PyTorch, etc.) and oneDNN is already integrated into them hence you will get speed-up from oneDNN optimizations and flexibility of DNN frameworks. If you still want to implement it using oneDNN directly then all those missing pieces should be implemented in your application.

**Comment by vpirogov** [comment]
Closing as stale.
